custom_model_runner/datarobot_drum/drum/language_predictors/r_predictor/r_predictor.py

```diff
@@ -181,4 +181,42 @@ def _rlist_to_dict(rlist):
         return ret
 
     def _transform(self, **kwargs):
-        raise DrumCommonException("Transform feature is not supported for R")
+        input_binary_data = kwargs.get(StructuredDtoKeys.BINARY_DATA)
+        target_binary_data = kwargs.get(StructuredDtoKeys.TARGET_BINARY_DATA)
+        mimetype = kwargs.get(StructuredDtoKeys.MIMETYPE)
+        with capture_R_traceback_if_errors(r_handler, logger):
+            predictions = r_handler.outer_transform(
+                binary_data=ro.rinterface.NULL
+                if input_binary_data is None
+                else ro.vectors.ByteVector(input_binary_data),
+                target_binary_data=ro.rinterface.NULL
+                if target_binary_data is None
+                else ro.vectors.ByteVector(target_binary_data),
+                mimetype=ro.rinterface.NULL if mimetype is None else mimetype,
+                model=self._model,
+            )
+
+        with localconverter(ro.default_converter + pandas2ri.converter):
+            py_data_object = ro.conversion.rpy2py(predictions)
+
+        if not isinstance(py_data_object, ro.ListVector) or len(py_data_object) != 2:
+            error_message = "Expected transform to return a two-element list containing X and y, got {}. ".format(
+                type(py_data_object)
+            )
+            logger.error(error_message)
+            raise DrumCommonException(error_message)
+
+        output_X = py_data_object[0]
+        output_y = py_data_object[1] if py_data_object[1] is not ro.NULL else None
+
+        if not isinstance(output_X, pd.DataFrame):
+            error_message = (
+                "Expected predictions type: {}, actual: {}. "
+                "Are you trying to run binary classification without class labels provided?".format(
```

Comment:
Did you mean to have this error message?

Reply:
removed the label part